### ISSUE TITLE
Error reading workspace when the remote workspace has the terraform version set to a constraint

### ISSUE THREAD

**Issue by kpenaseveriche** [hashicorp/go-tfe#994]
#### go-tfe version

The version used in [terraform 1.9.8](https://github.com/hashicorp/terraform/blob/v1.9.8/go.mod#L31):

```plaintext
1.58.0
```

## Description

When running commands that call workspace#Read, like `terraform untaint` (given that the current project is configured to use a remote workspace). If the remote TFE workspace has a version constraint set on the terraform-version field (which is supported as documented in the [API docs](https://developer.hashicorp.com/terraform/cloud-docs/api-docs/workspaces#create-a-workspace)). It produces an error about a malformed version.

```
Error: Error looking up workspace

Invalid Terraform version: Malformed version: ~> 1.6.0
```

## Testing plan

* Create a TFE workspace with a version constraint
* Set up a project configured with that workspace as the backend
* Apply a plan such that a resource exists
* Run `terraform untaint <resource name>`

#### Expected Behavior

Command edits the remote state.

#### Actual Behavior

Error looking up workspace

#### Additional Context

In our use-case, the remote workspace was created via the API, not the UI

**Comment by Maed223** [comment]
I was able to reproduce this error using the `backend "remote"` block, though thankfully the problem isn't present using the `cloud` block. This doesn't seem to be a `go-tfe` problem but rather a gap between TFE and the remote block. I'd recommend migrating to the `cloud` block as it is the documented best practice.

**Comment by sebasslash** [comment]
Hi @kpenaseveriche, this is a bug with the remote backend resolving workspace version constraints. Since this is a bug with a Terraform backend rather than go-tfe, I will go ahead and close this ticket. We are tracking this bug internally so the fix should be made available in a future `v1.10.X` Terraform release. If possible the recommendation, as @Maed223 pointed out, would be to migrate from the remote backend to the cloud backend, which should immediately resolve your issue.
